### ISSUE TITLE
[raysgd][tfrunner] Getting stuck after running the task 

### ISSUE THREAD

**Issue by SanjiAp** [ray-project/ray#8695]
[raysgd][tfrunner]
I am using tf runner example code and created a cluster with 2 different machines(192.168.140.44, 49)
After running "$ python sample_code.py --address 192.168.140.44:6379 -n 2"
I am getting the following log and in the end it stuck there for infinite time.
```
- 2020-06-01 11:43:48,179 WARNING worker.py:809 -- When connecting to an existing cluster, _internal_config must match the cluster's _internal_config.
- (pid=30968) 2020-06-01 11:43:50.643620: I tensorflow/core/platform/cpu_feature_guard.cc:143] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 FMA

- (pid=35845, ip=192.168.140.44) 2020-06-01 11:43:54.718155: I tensorflow/core/platform/cpu_feature_guard.cc:143] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 FMA
- (pid=35845, ip=192.168.140.44) 2020-06-01 11:43:54.730408: I tensorflow/core/platform/profile_utils/cpu_utils.cc:102] CPU Frequency: 2597005000 Hz
- (pid=35845, ip=192.168.140.44) 2020-06-01 11:43:54.738331: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x55b96458ee90 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
- (pid=35845, ip=192.168.140.44) 2020-06-01 11:43:54.738393: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version
- (pid=35845, ip=192.168.140.44) 2020-06-01 11:43:54.738530: I tensorflow/core/common_runtime/process_util.cc:147] Creating new thread pool with default inter op setting: 56. Tune using inter_op_parallelism_threads for best performance.
- (pid=35845, ip=192.168.140.44) E0601 11:43:54.747657173   35845 socket_utils_common_posix.cc:222] check for SO_REUSEPORT: {"created":"@1590992034.747646644","description":"SO_REUSEPORT unavailable on compiling system","file":"external/com_github_grpc_grpc/src/core/lib/iomgr/socket_utils_common_posix.cc","file_line":190}
- (pid=30968) 2020-06-01 11:43:50.655770: I tensorflow/core/platform/profile_utils/cpu_utils.cc:102] CPU Frequency: 2597035000 Hz
- (pid=30968) 2020-06-01 11:43:50.659451: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x562568aa1470 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
- (pid=30968) 2020-06-01 11:43:50.659495: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version
- (pid=30968) 2020-06-01 11:43:50.659704: I tensorflow/core/common_runtime/process_util.cc:147] Creating new thread pool with default inter op setting: 56. Tune using inter_op_parallelism_threads for best performance.
- (pid=30968) E0601 11:43:50.668888712   30968 socket_utils_common_posix.cc:222] check for SO_REUSEPORT: {"created":"@1590992030.668871022","description":"SO_REUSEPORT unavailable on compiling system","file":"external/com_github_grpc_grpc/src/core/lib/iomgr/socket_utils_common_posix.cc","file_line":190}
- (pid=30968) 2020-06-01 11:43:50.675626: I tensorflow/core/distributed_runtime/rpc/grpc_channel.cc:301] Initialize GrpcChannelCache for job worker -> {0 -> localhost:14724, 1 -> 192.168.140.44:54153}
- (pid=30968) 2020-06-01 11:43:50.679194: I tensorflow/core/distributed_runtime/rpc/grpc_server_lib.cc:390] Started server with target: grpc://localhost:14724
- (pid=35845, ip=192.168.140.44) 2020-06-01 11:43:54.762549: I tensorflow/core/distributed_runtime/rpc/grpc_channel.cc:301] Initialize GrpcChannelCache for job worker -> {0 -> 192.168.140.49:14724, 1 -> localhost:54153}
- (pid=35845, ip=192.168.140.44) 2020-06-01 11:43:54.769828: I tensorflow/core/distributed_runtime/rpc/grpc_server_lib.cc:390] Started server with target: grpc://localhost:54153
```

Ray version : 0.8.5
python version : 3.6.10
tensorflow version : 2.2.0
OS : centos 7 
*Ray version and other system information (Python version, TensorFlow version, OS):*

**Comment by SanjiAp** [comment]
Code is as follows

`

	import argparse
	import tensorflow as tf
	from tensorflow.keras.models import Sequential
	from tensorflow.keras.layers import Dense
	import numpy as np
	import json
	import ray
	#from ray import tune
	from ray.util.sgd.tf.tf_trainer import TFTrainer, TFTrainable
	import os
	import time
	#NUM_TRAIN_SAMPLES = 100000000
	NUM_TRAIN_SAMPLES = 1000000
	NUM_TEST_SAMPLES = 4000


	def create_config(batch_size):
		return {
			# todo: batch size needs to scale with # of workers
			"batch_size": batch_size,
			"fit_config": {
				"steps_per_epoch": NUM_TRAIN_SAMPLES // batch_size
			},
			"evaluate_config": {
				"steps": NUM_TEST_SAMPLES // batch_size,
			}
		}

	def linear_dataset(a=2, size=1000):
		x = np.random.rand(size)
		y = x / 2

		x = x.reshape((-1, 1))
		y = y.reshape((-1, 1))

		return x, y

	 def simple_dataset(config):
		batch_size = config["batch_size"]
		x_train, y_train = linear_dataset(size=NUM_TRAIN_SAMPLES)
		x_test, y_test = linear_dataset(size=NUM_TEST_SAMPLES)

		train_dataset = tf.data.Dataset.from_tensor_slices((x_train, y_train))
		test_dataset = tf.data.Dataset.from_tensor_slices((x_test, y_test))
		train_dataset = train_dataset.shuffle(NUM_TRAIN_SAMPLES).repeat().batch(
			batch_size)
		test_dataset = test_dataset.repeat().batch(batch_size)

		return train_dataset, test_dataset



	 def simple_model(config):
		model = Sequential([Dense(10, input_shape=(1, )), Dense(1)])

		model.compile(
			optimizer="sgd",
			loss="mean_squared_error",
			metrics=["mean_squared_error"])

		return model

	def train_example(num_replicas=1, batch_size=128, use_gpu=False):
		trainer = TFTrainer(
			model_creator=simple_model,
			data_creator=simple_dataset,
			num_replicas=num_replicas,
			use_gpu=use_gpu,
			verbose=True,
			config=create_config(batch_size))

		# model baseline performance
		start_stats = trainer.validate()
		print(start_stats)
		num_epochs = 10
		start = time.time()
		# train for 2 epochs
		for e in range(num_epochs):
			trainer.train()
		#trainer.train()
		stop = time.time()

		# model performance after training (should improve)
		end_stats = trainer.validate()
		print(end_stats)

		# sanity check that training worked
		dloss = end_stats["validation_loss"] - start_stats["validation_loss"]
		dmse = (end_stats["validation_mean_squared_error"] -
				start_stats["validation_mean_squared_error"])
		print(f"dLoss: {dloss}, dMSE: {dmse}")
		print("total time ray tf: ",stop - start)
		ray_time = stop - start
		sample_config = create_config(batch_size)

		tf_default_model = simple_model(sample_config)
		train_dataset, test_dataset = simple_dataset(sample_config)
		start = time.time()


		tf_default_model.fit(train_dataset, epochs=num_epochs, steps_per_epoch = sample_config["fit_config"]["steps_per_epoch"])
		print("total time tf default : ", time.time() - start)
		print(tf_default_model.evaluate(test_dataset,steps = sample_config["evaluate_config"]["steps"]))
		print("total time ray :",ray_time)

		if dloss > 0 or dmse > 0:
			print("training sanity check failed. loss increased!")
		else:
			print("success!")


	def tune_example(num_replicas=1, use_gpu=False):
		config = {
			"model_creator": tune.function(simple_model),
			"data_creator": tune.function(simple_dataset),
			"num_replicas": num_replicas,
			"use_gpu": use_gpu,
			"trainer_config": create_config(batch_size=128)
		}

		analysis = tune.run(
			TFTrainable,
			num_samples=2,
			config=config,
			stop={"training_iteration": 2},
			verbose=1)

		return analysis.get_best_config(metric="validation_loss", mode="min")


	if __name__ == "__main__":
		parser = argparse.ArgumentParser()
		parser.add_argument(
			"--address",
			required=False,
			type=str,
			help="the address to use for Ray")
		parser.add_argument(
			"--num-replicas",
			"-n",
			type=int,
			default=1,
			help="Sets number of replicas for training.")
		parser.add_argument(
			"--use-gpu",
			action="store_true",
			default=False,
			help="Enables GPU training")
		parser.add_argument(
			"--tune", action="store_true", default=False, help="Tune training")
		parser.add_argument(
			"--num_cpu",
			"-ncpus",
			type=int,
			default=1,
			help="Sets number of cpus for training.")
		parser.add_argument(
			"--batch_size",
			"-batch_size",
			type=int,
			default=128,
			help="Sets number of cpus for training.")


		args, _ = parser.parse_known_args()

		ray.init(address=args.address)#num_cpus=args.num_cpu)
               train_example(num_replicas=args.num_replicas, use_gpu=args.use_gpu, 
                  batch_size=args.batch_size)



`

**Comment by richardliaw** [comment]
Hm, is it possible that you're requesting more resources than you have on the cluster?

**Comment by SanjiAp** [comment]
> Hm, is it possible that you're requesting more resources than you have on the cluster?

If I am requesting more resources then ray gives different warning and stops there. I don't have screenshot of that right now.

I guess this issue was something to do with the firewall. I have cleared this issue.
Thanks
